junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java

```diff
@@ -997,11 +998,26 @@ public static List<Class<?>> findAllClassesInClasspathRoot(URI root, Predicate<C
 		return findAllClassesInClasspathRoot(root, ClassFilter.of(classNameFilter, classFilter));
 	}
 
+	/**
+	 * @see org.junit.platform.commons.support.ReflectionSupport#findAllClassesInClasspathRootAsStream(URI, Predicate, Predicate)
+	 */
+	public static Stream<Class<?>> findAllClassesInClasspathRootAsStream(URI root, Predicate<Class<?>> classFilter,
+			Predicate<String> classNameFilter) {
+		return findAllClassesInClasspathRootAsStream(root, ClassFilter.of(classNameFilter, classFilter));
+	}
+
 	/**
 	 * @since 1.1
 	 */
 	public static List<Class<?>> findAllClassesInClasspathRoot(URI root, ClassFilter classFilter) {
-		return Collections.unmodifiableList(classpathScanner.scanForClassesInClasspathRoot(root, classFilter));
+		return findAllClassesInClasspathRootAsStream(root, classFilter).collect(toUnmodifiableList());
+	}
+
+	/**
+	 * @since 1.1
+	 */
+	public static Stream<Class<?>> findAllClassesInClasspathRootAsStream(URI root, ClassFilter classFilter) {
+		return classpathScanner.scanForClassesInClasspathRoot(root, classFilter).stream();
```

Comment:
In this case, I would leave `findAllClassesInClasspathRoot` as it is, and call it here like this:
```suggestion
		return findAllClassesInClasspathRoot(root, classFilter).stream();
```

Otherwise, you would have one unnecessary "list -> stream -> list" conversion when calling `findAllClassesInClasspathRoot`.

Reply:
Same remark for `findAllClassesInPackage`,  `findAllClassesInModule` and `findNestedClasses`